design/ceph/osd-migration.md

```diff
@@ -36,139 +24,102 @@ considered in the future:
   status:
    storage:
       osd:
-       storeType:
-        bluestore: 3
-        bluestore-rdr: 5
+       migrationStatus:
+        completed: 3
+        pending: 5
   ```
 
-  - `storeType.bluestore`: Total number of BlueStore OSDs running
-  - `storeType.bluestore-rdr`: Total number of BlueStore-rdr OSDs running
+  - `osd.migrationStatus.completed`: Total number of OSDs that got migrated.
+  - `osd.migrationStatus.pending`: Total number of OSDs that are pending migration.
   - The cluster's `phase` should be set to `progressing` while OSDs are migrating
 
-The migration process will involve destroying existing OSDs one by one, wiping
-the drives, deploying a new OSD with the same ID, then waiting for all PGs
-to be `active+clean` before migrating the next OSD. Since this operation
-involves possible impact or downtime, users should be exercise caution
-before proceeding with this action.
-
-**NOTE**: Once the OSDs are migrated to a new backend, say `bluestore-rdr`, they
-won't be allowed to be migrated back to the legacy store (BlueStore).
-
-- Add a new label `osd-store:<osd store type>` to all OSD pods.
-- This label will help to identify the current backend being used for the OSD
-  and enable the operator to determine if the OSD should be migrated if the
-  the admin changes the backend type in the spec (`spec.Storage.store.type`).
-
-### New OSDs
-
-- Set the OSD backend provided by the user in `spec.storage.store.type` as an
-  environment variable in the OSD prepare job. If no OSD store is provided in
-  the spec, then set the environment variable to `bluestore`.
-- The prepare pod will use this environment variable when preparing OSDs
-  with the `ceph-volume` command.
-
-#### OSD Prepare
-
-  - RAW MODE
-
-    ```
-    ceph-volume raw prepare <OSD_STORE_ENV_VARIABLE> --data /dev/vda
-    ```
-
-  - LVM MODE
-
-    ```
-    ceph-volume lvm prepare <OSD_STORE_ENV_VARIABLE> --data /dev/vda
-    ```
-
-#### OSD Activation
-
-The ceph-volume `activate` command doesn't require the OSD backend to be passed
-as an argument. It auto-detects the backend that was used during when the OSD
-was prepared.
-
-### Existing OSDs
-
-- After upgrading an existing Rook Ceph cluster to a Ceph release that supports
-  `bluestore-rdr`, admins can migrate `bluestore` OSDs to `bluestore-rdr`.
-- The backend of OSDs can not be overridden, so this update will require
-  the OSDs to be replaced one by one.
-- In order to migrate OSDs to use `bluestore-rdr`, admins must patch the
-  Ceph cluster spec as below:
-
-```yaml
-storage:
-  store:
-    type: bluestore-rdr
-    updateStore: yes-really-update-store
-```
 
-#### Replace OSD
-The operator's reconciler will replace one OSD at a time.
-A configmap will be used to store the OSD ID currently being migrated.
+### Automated OSD Migration Process
+- The operator's reconciler will migrate one OSD at a time.
+- A configmap will be used to store the OSD ID currently being migrated.
 OSD replacement steps:
```

Comment:
need a blank line before/after bullet list, or else the markdown rendering is not right
```suggestion

OSD replacement steps:

```